### DISCUSSION TITLE
Triggering resurvey for locations where brand left the country

### DISCUSSION THREAD

**Discussion by matkoniecz** [streetcomplete/StreetComplete#3083] [Ideas]
There are common cases where specific brand left given area (for example Tesco disappeared from Poland).

When this happens very large number of objects become outdated. And it is typical that some of them continue operating under a different brand, some become another shop or vacant.

One of potential ways of handling this is to have list of dead brands in a given area. Some lists exists already (see #1354), but using them is tricky

- sometimes some shops actually remain (especially on border crossing where technically it may be in country A, but is treated as being in country B)
- sometimes brand may come back
- sometimes independent shops continue operating using old corporate branding
- sometimes reported closing of all shops does not actually involve closing all shops in this area

Either way there would be a need to somehow record "this shop is gone according to index of dead brands, but it is actually alive". We have no such tag.

So I want to propose new tag `object_likely_gone=brand_Tesco_left_this_area`. 

Such tags would be set automatically (with respecting https://wiki.openstreetmap.org/wiki/Automated_Edits_code_of_conduct ) on report that some brand is no longer present in a given area.

StreetComplete and other software could strongly suggest resurveying objects with `object_likely_gone` tag. In case of StreetComplete a resurvey quest would be triggered.

I want to propose this tag to a wider community (as discussion on issue tracker is never sufficient to push a new tag), but first I want to check is it something that makes sense on a StreetComplete side and are there obvious improvements.

Why not `fixme`? `fixme` is a freeform tag and parsing that is quite doomed and unlikely to work well. But maintaining list of `fixme` values that should trigger resurvey quest is also something that could be done. But it would be very annoying and prone to failures.

**Comment by Cj-Malone** [comment]
If this is gonna be a tag proposal it might be worth making it a little more generic, so you could have an option like `quality=bad_location` as well as `quality=object_likely_gone`.

Although having just read through https://github.com/streetcomplete/StreetComplete/issues/1354 I like the idea of StreetComplete (or third party) maintaining a list of these dead brands that then could be used to trigger a variant of the vacant quest.

**Reply by matkoniecz** [reply]
The problem with dead brand list is that it requires somehow marking the exceptions.

And at least I prefer special tag put on almost certainly bad OSM data, rather than tags put on valid OSM data just to mark that the external database is wrong.

**Reply by Cj-Malone** [reply]
I don't think it's out of the question to have a list of exceptions with the dead brand list, or it being excluded based on a `check_date`. However I do usually argue for more things to be in database than out, so maybe a tag is better. I can imagine data consumers ignoring objects with `object_likely_gone` older than `x`.

**Reply by matkoniecz** [reply]
> I don't think it's out of the question to have a list of exceptions with the dead brand list

But how StreetComplete answer would end there? Calling third-party API?

**Reply by mnalis** [reply]
I agree with other part of @Cj-Malone answer,  that using `check_date` (or even better, `survey:date`) would be better way to mark exceptions in such case. So one could check the list of dead brands in the area, and mark for solving only those which do not have `survey:date` newer than creation date of that list. (I'd still prefer `fixme` with hashtags, though)

**Reply by Cj-Malone** [reply]
@matkoniecz If that's the way it goes, I'd put my name down to work on that dataset and API. Although there is a few things to think about:

1. This is getting closer to MapRoulette/osmose. Would it make more sense to use that as a quest source, obviously not only select challenges. StreetComplete would send "Fixed" or "Not an Issue" back to MapRoulette where the challenge creator can curate and validate things. @mvexel
2. Is it even necessary to record them? With your example of Tesco in Poland, there is a 18-month transition period from 2020-06, so by 2022 everything should be changed. If it gets re asked to people in that period is it really an issue? and actually the second time it's asked it is more likely to have changed.
3. StreetComplete could bump a tag like `check_date:brand` to avoid re asking and the need of a third party API. I would argue that the list of dead brands should still be external to StreetComplete so other editors can utilise it.

**Comment by mnalis** [comment]
I'm wondering if it might not be better to use some hashtag (like we have `#surveyme`) in the regular `fixme` tag? eg.
`fixme=#surveyme #likelygone  brand_Tesco_left_this_area`

That way:
- it would automatically work in all the tools that support `fixme`
- it allows for easy filtering via `#likelygone`
- it should not be prone to false positives (eg. users are unlikely to type `#likelygone` by mistake)
- it could work in StreetComplete for this, and more generally due to `#surveyme` if `fixme` handling [is implemented](https://github.com/streetcomplete/StreetComplete/issues/3084)
- it does not require new tag
- extensions for similar ideas are similarly easy to implement without changes to the tools (eg. https://github.com/streetcomplete/StreetComplete/issues/1585, https://github.com/streetcomplete/StreetComplete/discussions/2477#discussioncomment-439407 etc)

**Reply by matkoniecz** [reply]
Hm, I like this idea despite my massive dislike of hashtags in general.

**Comment by mnalis** [comment]
If new tag is to be invented, then perhaps at least put it in `fixme` prefix namespace, eg. `fixme:object_likely_gone=brand_Tesco_left_this_area` or `fixme:quality=bad_location`

That would at least allow the tools to easily automagically support this and other similar tags that may pop up.

**Comment by Cj-Malone** [comment]
Looks like NSI already documents [dissolved brands](https://github.com/osmlab/name-suggestion-index/blob/main/dist/dissolved.json), so a potential solution is to pull them into the check existence quest. Where they could be changed into a new shop, vacant shop or bump the survey date.

**Reply by matkoniecz** [reply]
How one would handle where NSI is wrong and object exists? How SC would mark "ignore NSI for this object"?

There are some shops that continued to operate under brand no longer existing in a given area (they never changed signpost/decorations) and weird cases like airports and other special areas.

**Reply by Cj-Malone** [reply]
`check_date:brand` or `survey:date`?

**Reply by matkoniecz** [reply]
And skipping all objects with `check_date:brand`? This could work but would be a bit cryptic.

**Reply by Cj-Malone** [reply]
Well it'd be resurveyed every x like the other `check_date` quests.

Ah, I see, specifically with the permanent exceptions. I don't have a solution, other than keeping them in the resurvey, but if x is reasonably large I don't think it'd be a massive issue to keep asking it indefinitely.
